doc/adr/0003-self-hosted.md

```diff
@@ -0,0 +1,86 @@
+# 3. Self Hosted
+
+Date: 2021-04-27
+
+## Status
+
+Proposed
+
+## Context
+
+We want to enable enterprise customers to self host Athens knowledge graph on premise.  
+Uses should be able to collaborate on same graph, edit block & pages as well as see presence of other collaborators.
+
+## Decision
+
+We need a backend server that at least talks Datalog.
+
+For 1st backend we've chosen Datahike, because it's closeness to Datascript that we already use.  
+Another possible backend could be Crux or Datomic.
+
+### Communication Layer
+
+Because we're running in a very controlled environment (as in we don't have to support Samsung Internet & FB embedded browser)  
+we can assume availability of modern Web APIs, so for client server communication we can rely purely on WebSockets.
+
+### Concurrency
+
+To avoid problems of concurrent updates, we'd like to have single writer approach.  
+This means that updates to central graph can be done synchronously.
+
+### Connecting clients
```

Comment:
this does not seem to mention any way to verify the client is allowed to access the server at all. we should have at least some simple security for that. shared secret would be enough for now. we just shouldn't allow anyone who can access the IP & port to read/write.